pkg/services/pluginsintegration/clientmiddleware/caching_middleware.go

```diff
@@ -77,7 +90,12 @@ func (m *CachingMiddleware) QueryData(ctx context.Context, req *backend.QueryDat
 
 	// Update the query cache with the result for this metrics request
 	if err == nil && cr.UpdateCacheFn != nil {
-		cr.UpdateCacheFn(ctx, resp)
+		// If AWS async cahing is enabled and it is a still running async query, don't cache it
+		if m.features == nil || !m.features.IsEnabled(featuremgmt.FlagAwsAsyncQueryCaching) {
```

Comment:
I think I don't understand why we call update cache when the feature flag is off, isn't that the opposite of the comment? am I misunderstanding something?

Reply:
If the feature toggle is off, we always want to call the UpdateCacheFn. Otherwise, we only want to call it if the shouldCache func returns true. The logic is right, but the comment could be a tad clearer.

Reply:
If the feature flag is off, our async queries won't (shouldn't) hit this path at all. If the flag is off, we attach a Cache-Skip header to all the status checks (see [here](https://github.com/grafana/grafana-async-query-data-js/pull/17/files)).
What this is actually guarding is that if the flag is off, the caching service runs the way it always has (without calling `awsds.ShouldCacheQuery`) in the off case that we've added a bug there.

Reply:
Redid the comment, hopefully clearer now

Reply:
i see makes sense thanks!